bundles/framework/layerselector2/Flyout.js

```diff
@@ -26,7 +26,12 @@ Oskari.clazz.define('Oskari.mapframework.bundle.layerselector2.Flyout',
 
         this.mapLayerService = Oskari.getSandbox().getService('Oskari.mapframework.service.MapLayerService');
         this.layerlistService = Oskari.getSandbox().getService('Oskari.mapframework.service.LayerlistService');
-
+        
+        this.mapLayerService.on('Layers.Loaded', function (loaded) {
```

Comment:
Remove this event and use MapLayerEvent instead. Also this is only sent when the bulk loading is completed. Currently it doesn't react to runtime changes on the layers. Using MapLayerEvent will solve that as well (if layer id is null -> it's a massive update like this and when layer id is present only the layer referenced changed).